### ISSUE TITLE
Implement a "Shared" counter

### ISSUE THREAD

**Issue by phillipadsmith** [TheTyee/design-article.thetyee.ca#319]
[Per Basecamp thread](https://basecamp.com/1929296/projects/9481270/messages/60066584#comment_443149247), the request is to implement a count of shares across networks and display that on article pages, presumably near the share tools.

@robynleighsmith Can you put links to the examples we looked at in the office here?

**Comment by robynleighsmith** [comment]
I think we just looked at the Guardian and the CBC. Not sure how either are calculated.

In the two examples below, both of the share options are on the left. 

https://www.theguardian.com/world/2016/aug/02/chlorine-attack-syria-dozens-ill-saraqeb-idlib

http://www.cbc.ca/news/world/obama-trump-1.3704157

**Comment by MrBryan** [comment]
OK here it is in testing mode (should show up in any story you newly test with the numbers that actually will show if it went live.

The free version is limited to 10,000 views per day or something so we'd need the $40 a month version that does up to  200,000 ( we get around 30,000 pageviews from memory..)

http://preview.thetyee.ca/Opinion/2016/08/05/Farewell-to-Mel-Hurtig/

![image](https://cloud.githubusercontent.com/assets/1721891/17450682/afc97a80-5b2f-11e6-8797-e7a000b877c7.png)

**Comment by MrBryan** [comment]
We could add the count of the email to friend likely with a little effort if that is significant - currently it's counting facebook (shares - likes also available), twitter, linkedin, reddit.

**Comment by robynleighsmith** [comment]
Thanks Bryan. To be crystal clear, what does this version count, specifically? Just Facebook shares? Does it cost more to count likes? And to include the other share numbers?

**Comment by MrBryan** [comment]
Nope it costs the same just more code  to add facebook likes - it counts all that I listed above you probably typed that before my second comment :)

**Comment by MrBryan** [comment]
We have all the numbers separately I'm just adding them up -- could do something like this alternately: 
![image](https://cloud.githubusercontent.com/assets/1721891/17450906/02874e7c-5b31-11e6-9ebb-97960f270dfa.png)

**Comment by MrBryan** [comment]
and when i say newly test, I mean that if you preview a story from bricolage the number should show up now

**Comment by MrBryan** [comment]
the number for the live version of the story

**Comment by MrBryan** [comment]
it breaks the toolbar onto two lines for mobile so that will have to be tweaked just think of this as a draft for comment

**Comment by robynleighsmith** [comment]
Okay, so the current number being previewed shows Facebook likes, shares, Google +, and whatever "Submit" is? It doesn't include comments, obviously.

Also, it doesn't show up on this preview: http://preview.thetyee.ca/Opinion/2016/08/02/BC-Terror-Trial-Verdict/

**Comment by robynleighsmith** [comment]
Put another way, the current number showing in preview is the sum total of all those things?

**Comment by robynleighsmith** [comment]
Also, there's no free version of this? I think we got Facebook shares shown for free before the redesign.

**Comment by MrBryan** [comment]
oh no that "submit" thing is just a picture from HuffPo's share bar that means nobody has posted to reddit yet instead of showing zero.  We are combining facebook , twitter, reddit, linkedin, google+. 

I thought this one had addressed the twitter thing but it seems not so perhaps it is not worth the fee (twitter counts don't work since twitter turned off access to that data.)  I thought I remembered a directive saying let's go with sharedcount for now is why I started there.

There are people who provide the service of twitter counts presumably so that they can mine your data that we could investigate if that is important:  

http://twitcount.com/
https://opensharecount.com/

And there is probably a free way of getting those (non-twitter) counts, either by coding myself or using a third party toolbar provider (normally they provide you with sharing statistics and also mine your data of course) like http://www.addthis.com/

**Comment by robynleighsmith** [comment]
Okay, so this "SharedCount" that you're previewing now includes Facebook shares and likes, reddit, linkedin and google+. Not Twitter, because no one can get Twitter. And it will cost $40 a month because we have received more than 10,000 views daily.

Do I have this right? (sorry if this is like hitting your head against a brick wall repeatedly).

**Comment by MrBryan** [comment]
Yes that is right except that some people are claiming they can approximate twitter- twitcount and opensharecount among others - they are free so could experiment or investigate more if we find that twitter is important to have in the mix.  

That aside, If we want to avoid the $40 fee I can look at how to look up each of those numbers ourselves in code rather than use their simplified service, probably will pay off in the long run just whether to spend the time now is the question.,

**Comment by robynleighsmith** [comment]
Well, we may as well use this solution for now (including paying the $40 fee), as it will probably take time to code our own version. Thanks for the investigation, Bryan. I'll just wait to give @phillipadsmith to say his piece before giving the official go-ahead.

**Comment by phillipadsmith** [comment]
Wow, this is a crazy-long thread... :-/ 

Few quick thoughts:
1. Agree that we should move forward with what is easiest for the moment, including the $40 fee, which is very low compared to development time costs.
2. It would be easy enough to aggregate the counts we care about and run that on The Tyee's infrastructure for free within a year, e.g., focus on just Facebook, Twitter, and Reddit if that's interesting.
3. We should not go down the ShareThis/AddThis path again because we know that it A) impacts the load time of the site, and B) doesn't count accurately (counts button clicks, not actual shares), and C) looks super ugly. Right now, the custom buttons let Eric track interactions closely in Google Analytics, which is what we wanted. 
4. I'd propose we should throw in the e-mail share count because we know that LOTS of Tyee readers use that tool, and the data is easy to get, e.g., https://github.com/phillipadsmith/widgets.thetyee.ca   <-- BTW, Bryan, it looks like the widgets.thetyee.ca service is down for some reason (probably the load the other day), would  you be so kind as to re-start it?

Over and out for now. :-)

**Comment by robynleighsmith** [comment]
All right, it's agreed. For now, we moved ahead with Shared Count, which includes the gamut of shares except for Twitter. (@MrBryan, let me know if you need me to input credit card data for the $40 somewhere.) 

@phillipadsmith, I'm assuming the email share count would not be included in the total Shared Count counter, but rather in a separate way?

**Comment by phillipadsmith** [comment]
> @phillipadsmith, I'm assuming the email share count would not be included in the total Shared Count counter, but rather in a separate way?

I believe it would be easy enough to include the e-mail share count in that single number, as well as Twitter.

**Comment by robynleighsmith** [comment]
Let's do it then!

**Comment by MrBryan** [comment]
OK Sent the login info your email @robynleighsmith  .  Alex offered to do the styling for this although not this week. If we want to get it out sooner I can work something out quickly in the interim

**Comment by MrBryan** [comment]
I should say _pehaps_ not this week --- alex?

**Comment by robynleighsmith** [comment]
Hm, styling? This is the first I'm hearing of it. Yes, would be good to get a handle on how long that will take. 

@phillipadsmith, getting this done is still the priority, yes? Meaning styling the share counters should be a top priority for Alex, too.

**Comment by MrBryan** [comment]
Yes just some thinking has to be done about what should happen at mobile sizes - the bar is too wide with this element added

![image](https://cloud.githubusercontent.com/assets/1721891/17497706/3bfa1092-5d91-11e6-9f0c-24acbe8e069c.png)

**Comment by robynleighsmith** [comment]
I would like this implemented ASAP. Can't wait another two weeks for styling. Can we implement now and then add Alex's stylings later?

**Comment by MrBryan** [comment]
Definitely, i'll get it working tomorrow a.m

**Comment by MrBryan** [comment]
OK it's all set  

![image](https://cloud.githubusercontent.com/assets/1721891/17524440/a2618392-5e2c-11e6-8745-e5482e4a2085.png)

Just the sharedcount stuff for now (no twitter + email shares added in yet)

It removes the word "shares" and "join" for the mobile width on the toolbar

http://preview.thetyee.ca/Opinion/2016/08/05/Farewell-to-Mel-Hurtig/

Let me know if that is all good and I'll deploy it to live

**Comment by phillipadsmith** [comment]
> It removes the word "shares" and "join" for the mobile width on the toolbar

Great interim solution, @MrBryan, many thanks for that. :-)

**Comment by phillipadsmith** [comment]
Also, side note, what we're asking @alexgreen to think through in terms of styling is _both_ this "Shares" data, as well as the "X comments" data (which would be either "X comments" or "No comments yet").

**Comment by robynleighsmith** [comment]
Looks great, @MrBryan. Ready to deploy. Looking forward to @alexgreen's treatment.

**Comment by MrBryan** [comment]
Ok she is live

**Comment by MrBryan** [comment]
!

**Comment by robynleighsmith** [comment]
Thanks!

**Comment by phillipadsmith** [comment]
Research from today:

Two short-term solutions:

Twitter...
http://public.newsharecounts.com/count.json?url=http://thetyee.ca/Opinion/2016/08/05/Protecting-BC-Rights/

Facebook...
http://api.facebook.com/restserver.php?format=json&method=links.getStats&urls=http://thetyee.ca/Opinion/2016/08/05/Protecting-BC-Rights/

Getting it done "right" solutions:

Object Graph (signed with OAuth token)
https://developers.facebook.com/docs/graph-api/reference/v2.1/object/likes/
https://developers.facebook.com/tools/explorer/145634995501895/?method=GET&path=%3Fid%3Dhttp%3A%2F%2Fthetyee.ca%2FOpinion%2F2016%2F08%2F05%2FProtecting-BC-Rights%2F&version=v2.1

```
{
  "og_object": {
    "id": "1150740568332297",
    "description": "Two years after Mount Polley, questions about gov’t decisions linger.",
    "title": "Where’s the Will to Protect British Columbians’ Rights after Resource Disasters? | The Tyee",
    "type": "article",
    "updated_time": "2016-08-06T03:49:10+0000",
    "url": "http://thetyee.ca/Opinion/2016/08/05/Protecting-BC-Rights/"
  },
  "share": {
    "comment_count": 0,
    "share_count": 499
  },
  "id": "http://thetyee.ca/Opinion/2016/08/05/Protecting-BC-Rights/"
}
```

And Twitter REST API search endpoint, which requires signing as well as paging.

**Comment by phillipadsmith** [comment]
@MrBryan:

Question... the current share on this article is 160:
<img width="822" alt="where s_the_will_to_protect_british_columbians _rights_after_resource_disasters____the_tyee" src="https://cloud.githubusercontent.com/assets/34128/17604646/8bbbbe8c-5fcb-11e6-8a01-63cf755cf24a.png">

But the data with sharedcount.com says 499:
http://www.sharedcount.com/#url=http%3A%2F%2Fthetyee.ca%2FOpinion%2F2016%2F08%2F05%2FProtecting-BC-Rights%2F
<img width="275" alt="sharedcount__social_url_analytics" src="https://cloud.githubusercontent.com/assets/34128/17604667/a089bd5a-5fcb-11e6-97ef-d547bc946a78.png">

Is the wrong property being used?

**Comment by robynleighsmith** [comment]
Argh!! I had a feeling that those numbers were off. Frustrating.

@MrBryan @phillipadsmith

**Comment by MrBryan** [comment]
What is wrong there? In the above the SHARES say 160 which is what we are measuring, not likes or comments

**Comment by MrBryan** [comment]
I had mentioned we could be tracking likes separately if we want

**Comment by MrBryan** [comment]
I re-read what you had asked above and I see that I did not notice that you later asked if likes were included, too.  I can change this to the "total" amount right now if you want

**Comment by MrBryan** [comment]
But perhaps we have to change the text that says "shares" or just remove it?

**Comment by phillipadsmith** [comment]
I think that we should use the count that we'll be moving to per above link to Object Graph API, e.g., http://www.sharedcount.com/#url=http%3A%2F%2Fthetyee.ca%2FOpinion%2F2016%2F08%2F05%2FProtecting-BC-Rights%2F

Matches what's being returned above:

```
{
  "og_object": {
    "id": "1150740568332297",
    "description": "Two years after Mount Polley, questions about gov’t decisions linger.",
    "title": "Where’s the Will to Protect British Columbians’ Rights after Resource Disasters? | The Tyee",
    "type": "article",
    "updated_time": "2016-08-06T03:49:10+0000",
    "url": "http://thetyee.ca/Opinion/2016/08/05/Protecting-BC-Rights/"
  },
  "share": {
    "comment_count": 0,
    "share_count": 499
  },
  "id": "http://thetyee.ca/Opinion/2016/08/05/Protecting-BC-Rights/"
}
```

Which is the "total" property. So, maybe "share" is the wrong term to be using. Maybe "engagements"?

**Comment by MrBryan** [comment]
That's too long maybe just blank like the site you posted, only uses an icon

**Comment by phillipadsmith** [comment]
> That's too long maybe just blank like the site you posted, only uses an icon

I like the little bonfire icon. Or the one you've got there now. Or maybe we could eventually swap the icon / icon colour based on the number of engagements. ;-)

**Comment by MrBryan** [comment]
haha.. OK first step, remove "shares" and use "total" number -- say OK if you are good with that Robyn, it will make all the stories jump immediately so that might require explanations or maybe they'll all just be happier ..

**Comment by robynleighsmith** [comment]
Yes, please, that's what I expected it to reflect all along! @MrBryan

**Comment by robynleighsmith** [comment]
Please let me know when it's done and live.

**Comment by MrBryan** [comment]
OK it's live now.  Sorry about the miscommunication :(

**Comment by robynleighsmith** [comment]
No worries, we persevered. Thanks much.

**Comment by alexgreen** [comment]
@robynleighsmith I think this may have been addressed already but is there a reason why the shares need to appear at the left side of the social toolbar? 
It might make more sense visually and semantically to group the metrics on the right side.

**Comment by robynleighsmith** [comment]
@alexgreen No reason that I can think of. Go with what you think will look best.

**Comment by phillipadsmith** [comment]
FYI, the new endpoints to retrieve aggregate share counts from The Tyee's own internal tools is now live. Documentation is [here](https://github.com/TheTyee/widgets.thetyee.ca#shares-by-url). Should be sufficient to re-implement the functionality of sharedcount.com.

**Comment by phillipadsmith** [comment]
These two commits:
- https://github.com/TheTyee/design-article.thetyee.ca/commit/dd986f303385760baafca2af3cc5b778958958ca
- https://github.com/TheTyee/design-article.thetyee.ca/commit/02d300f3b4dfa09f19e2e2e628230765e0aba435

Re-implement the shared count with the new service, e.g.,
http://thetyee.github.io/design-article.thetyee.ca/

Ready for deployment, @MrBryan

**Comment by MrBryan** [comment]
Deployed!

**Comment by robynleighsmith** [comment]
@phillipadsmith @MrBryan Looks good, though in mobile view the comments option gets bumped down a line, which doesn't look ideal. 

I forget what the fix was when this was happening before?

**Comment by MrBryan** [comment]
We were just discussing that.. we were saying perhaps see if anyone clicks on the join button from mobile and removing it but i just though of this - how about just removing the shared icon and leaving the count : 

![image](https://cloud.githubusercontent.com/assets/1721891/17840932/7a7ef4dc-67e0-11e6-9b50-ed00c4544dff.png)

**Comment by MrBryan** [comment]
we could do that just to avoid the line jumping for now then let alex weigh in if he wants

**Comment by robynleighsmith** [comment]
Sounds good.

**Comment by MrBryan** [comment]
ok that's live too

**Comment by phillipadsmith** [comment]
> Sounds good.

Looks a little disassociated to me, but I'm sure it's fine for the moment.

My current thought is: What about having the number sit on top of a faint icon? Obviously, for popular articles the number would float past the right-most edge of said icon, but I think it would still work.

Over to you, @alexgreen.

**Comment by phillipadsmith** [comment]
Could also be a typographic treatment, e.g.,: <img width="82" alt="twitter_counts_are_back__and_a_few_other_goodies_-_warfare_plugins" src="https://cloud.githubusercontent.com/assets/34128/17842083/f822ca38-67d3-11e6-8236-a2970cf067b8.png">

**Comment by robynleighsmith** [comment]
Oh Phillip, that looks great!!

**Comment by MrBryan** [comment]
With the icon still in there it doesn't save us much horizontal real-estate... what do you think of this:

![image](https://cloud.githubusercontent.com/assets/1721891/18178856/08842372-704e-11e6-9a14-c51eaac9824e.png)

http://preview.thetyee.ca/News/2016/09/01/Jims-Story-Holes-In-Downtown-Eastside-Systems/index.html

**Comment by MrBryan** [comment]
stilil works in the thousands (maybe give  tad more horizontal padding)

![image](https://cloud.githubusercontent.com/assets/1721891/18178941/51d6d0b0-704e-11e6-94fb-8be380624fcd.png)

**Comment by MrBryan** [comment]
Trying an option of floating the share icon to the left as you did phillip but outside of the official toolbar boundary:

![image](https://cloud.githubusercontent.com/assets/1721891/18179669/9c912102-7051-11e6-8a8d-a353646ec165.png)

**Comment by MrBryan** [comment]
to be refined later today just grabbing some late lunch!

**Comment by phillipadsmith** [comment]
Thank you for this @MrBryan! Personally, I prefer the option without the mini icon and just the text treatment. @robynleighsmith is offline this week, I believe, so we'll need to make the call and move forward and can always re-visit later.

Phillip.

**Comment by phillipadsmith** [comment]
Oh, and, I didn't mean to include the icon in the screenshot I sent for the reason you stated -- it would not save much space if it was there! Oversight on my part.

**Comment by robynleighsmith** [comment]
@phillipadsmith @MrBryan Look at me, back from the offline grave! I agree with Phillip and prefer the option without the icon. Text treatment works. Thanks!

**Comment by MrBryan** [comment]
OK she's live :)

**Comment by MrBryan** [comment]
... with better alignment with bottom and consistent spacing like the other icons to each other

**Comment by MrBryan** [comment]
![image](https://cloud.githubusercontent.com/assets/1721891/18184134/91593ed2-7065-11e6-9099-349903f7eee1.png)

**Comment by phillipadsmith** [comment]
Woo hoo!

And does that work acceptably on smaller screens now too?

Phillip.

**Comment by MrBryan** [comment]
yes except because of the print icon it overflows so going to hide that on mobile too

**Comment by phillipadsmith** [comment]
> yes except because of the print icon it overflows so going to hide that on mobile too
> 👍

**Comment by robynleighsmith** [comment]
@MrBryan @phillipadsmith I am so thrilled by how this turned out. Thank you both!!

**Comment by phillipadsmith** [comment]
LOL.

It’s Sunday, @robinleighsmith! Turn off the Internet. Go outside and play. ;-)

**Comment by MrBryan** [comment]
Speak for yourself, I prefer that my coworkers thank and praise me all weekend long

**Comment by MrBryan** [comment]
And of course you're welcome, it was fun as usual :)
